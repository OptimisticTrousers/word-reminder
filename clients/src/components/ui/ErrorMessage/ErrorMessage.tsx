import { CircleAlert } from "lucide-react";
import CSSModules from "react-css-modules";

import styles from "./ErrorMessage.module.css";

interface Props {
  message: string;
}

export const ErrorMessage = CSSModules(
  function ({ message }: Props) {
    return (
      <p styleName="message">
        <CircleAlert styleName="message_icon" />
        <span styleName="message_text">{message}</span>
      </p>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
