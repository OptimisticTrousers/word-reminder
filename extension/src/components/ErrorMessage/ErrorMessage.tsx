import { FC } from "react";
import CSSModules from "react-css-modules";
import { GoAlert } from "react-icons/go";
import styles from "./ErrorMessage.module.css";

interface Props {
  message: string;
}

const ErrorMessage: FC<Props> = CSSModules(
  ({ message }) => {
    return (
      <p styleName="message">
        <GoAlert styleName="message__icon" />
        <span styleName="message__text">{message}</span>
      </p>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "ignore" }
);

export default ErrorMessage;
