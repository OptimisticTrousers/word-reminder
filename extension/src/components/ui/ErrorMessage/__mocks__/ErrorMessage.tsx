export function ErrorMessage({ message }: { message: string }) {
  return <div data-testid="error-message">{message}</div>;
}
