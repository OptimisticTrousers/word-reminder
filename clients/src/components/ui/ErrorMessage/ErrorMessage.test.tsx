import { render, screen } from "@testing-library/react";
import { ErrorMessage } from "./ErrorMessage";

describe("ErrorMessage component", () => {
  it("renders message", () => {
    const props = { message: "Error Message." };

    const { asFragment } = render(<ErrorMessage {...props} />);

    const message = screen.getByText(props.message);
    expect(asFragment()).toMatchSnapshot();
    expect(message).toBeInTheDocument();
  });
});
