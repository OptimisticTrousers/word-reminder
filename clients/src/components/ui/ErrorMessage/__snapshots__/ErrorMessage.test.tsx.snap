// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage component > renders message 1`] = `
<DocumentFragment>
  <p
    class="_message_414cc6"
  >
    <svg
      class="lucide lucide-circle-alert _message_icon_414cc6"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <line
        x1="12"
        x2="12"
        y1="8"
        y2="12"
      />
      <line
        x1="12"
        x2="12.01"
        y1="16"
        y2="16"
      />
    </svg>
    <span
      class="_message_text_414cc6"
    >
      Error Message.
    </span>
  </p>
</DocumentFragment>
`;
